src/batch-client/index.ts

```diff
@@ -1335,16 +1335,27 @@ export class ZimbraBatchClient {
 			singleRequest: true
 		});
 
-	public modifyPrefs = (prefs: PreferencesInput) =>
-		this.jsonRequest({
+	public modifyPrefs = (prefs: PreferencesInput) => {
+
+		let attrs: any = mapValuesDeep(prefs, coerceStringToBoolean);
+		const {zimbraPrefMailTrustedSenderList} = prefs;
+
+		if (zimbraPrefMailTrustedSenderList &&
+			zimbraPrefMailTrustedSenderList instanceof Array &&
+			zimbraPrefMailTrustedSenderList.length === 0) {
+				attrs.zimbraPrefMailTrustedSenderList = '' ;
+		}
+
+		return this.jsonRequest({
 			name: 'ModifyPrefs',
 			namespace: Namespace.Account,
 			body: {
-				_attrs: mapValuesDeep(prefs, coerceBooleanToString)
+				_attrs: attrs
```

Comment:
previously we were doing coerceBooleanToString and now we are doing coerceStringToBoolean

Reply:
will update